### ISSUE TITLE
"TonemapShaderGLES2: Fragment shader compilation failed" on HTML5 with GLES2

### ISSUE THREAD

**Issue by dsnopek** [godotengine/godot#61167]
### Godot version

3.5.rc1

### System information

Firefox 99 and Chrome 101

### Issue description

When running a GLES2 game exported to HTML5 (by Godot 3.5-rc1) in either Firefox or Chrome, I get a whole bunch of errors, starting with "TonemapShaderGLES2: Fragment shader compilation failed"

This same game project doesn't have any issues when exported by Godot 3.4.4.

Here's a screenshot of the full set of error messages:

![Selection_384](https://user-images.githubusercontent.com/191561/169142324-69842835-1a79-40e1-b491-69d8108c880e.png)

### Steps to reproduce

~I've tried to reproduce with a fresh project, but I can't seem to. I'll keep trying to figure it out, but I decided to post this issue right away just in case the error message is enough for someone to guess what's going wrong.~

UPDATE:

* Create a fresh project
* Enable FXAA in project settings
* Export to HTML5
* Open in a web browser.

**Comment by clayjohn** [comment]
Likely also a regression from https://github.com/godotengine/godot/pull/54585

Should be as simple as making the following changes:
https://github.com/godotengine/godot/blob/b0a2b34d484dcf25fe87bc234b1ec264559c1da2/drivers/gles2/shaders/tonemap.glsl#L207
to 
```
 color.a = a + color.a * (1.0 - a); 
````
and 
https://github.com/godotengine/godot/blob/b0a2b34d484dcf25fe87bc234b1ec264559c1da2/drivers/gles2/shaders/tonemap.glsl#L245-L248
to
```
 float lumaNW = dot(rgbNW.rgb, luma) - ((1.0 - rgbNW.a) / 8.0); 
 float lumaNE = dot(rgbNE.rgb, luma) - ((1.0 - rgbNE.a) / 8.0); 
 float lumaSW = dot(rgbSW.rgb, luma) - ((1.0 - rgbSW.a) / 8.0); 
 float lumaSE = dot(rgbSE.rgb, luma) - ((1.0 - rgbSE.a) / 8.0); 
```
and
https://github.com/godotengine/godot/blob/b0a2b34d484dcf25fe87bc234b1ec264559c1da2/drivers/gles2/shaders/tonemap.glsl#L270
to
```
float lumaB = dot(rgbB.rgb, luma) - ((1.0 - rgbB.a) / 8.0); 
```

**Comment by dsnopek** [comment]
> Likely also a regression from https://github.com/godotengine/godot/pull/54585

Aha! Based on looking at that issue, I figured out what I was missing to reproduce: All you need to do is enable FXAA on a fresh project.

**Comment by akien-mga** [comment]
Fixed by #61171.
